pkg/controllers/binding/common.go

```diff
@@ -39,7 +40,7 @@ func ensureWork(
 
 	var jobCompletions []workv1alpha2.TargetCluster
 	var err error
-	if workload.GetKind() == util.JobKind {
+	if workload.GroupVersionKind() == batchv1.SchemeGroupVersion.WithKind(util.JobKind) {
```

Comment:
`GroupVersionKind` or `GroupKind`?

> I suppose that it's better to use GroupVersionKind().GroupKind() to replace these judgement.

Echo from https://github.com/karmada-io/karmada/issues/3011, I think what @Garrybest means is `GroupKind`.

Reply:
I'd like to user GroupVersionKind rather than GroupKind because we finally use the resource objects with specific version.
I'm afraid there will also be previous problems regardless of the version.  /cc @Garrybest for opinions.

Reply:
> Echo from #3011, I think what @Garrybest means is `GroupKind`.

+1, should we care about version? I thought that an object has different versions but we only care about group kind here. I guess we should handle both `batch/v1beta1` and `batch/v1` here.

Reply:
Take an extreme example:
```
if workload.GroupVersionKind() == batchv1.SchemeGroupVersion.WithKind(util.JobKind) {
        job := &batchv1.Job{}
	err := helper.ConvertToTypedObject(workload, job)
        ....
}
```
In many cases, we judge the kind of workload and convert it into specific version of resource(e.g. batchv1/job). If the two versions of a resource have obvious field changes, does it mean that the converted Job has missing content? We cannot guarantee that these missing fields will not affect subsequent logical judgments.

Reply:
> I guess we should handle both batch/v1beta1 and batch/v1 here.

> In many cases, we judge the kind of workload and convert it into specific version of resource

Both points make sense.
Changing `Kind` to `GroupKind` doesn't change anything, but changing to `GroupVersionKind` would reduce the range of supported API.

> If the two versions of a resource have obvious field changes, does it mean that the converted Job has missing content?

We should carefully evaluate every change. Please @Poor12 check if there are obvious field changes.

Reply:
Interesting, I realize the `GroupVersionKind` of workload comes from [preferred resources of detector](https://github.com/karmada-io/karmada/blob/d349f001db4551ffa814fcf78086798b270209c2/pkg/util/lifted/discovery.go#L39). So I guess if a preferred version is used here, e.g. `batch/v1` rather than `batch/v1beta1`, determination by GVK is fine.

Reply:
Yes, I remembered that for multi-version resources, now we will only create the preferred version of the Informer.